File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int rows=grid.size();
        int cols=grid[0].size();
        vector<vector<int>> res;
        for(int i=0;i<rows-2;i++){
            vector<int> temp;
            for(int j=0;j<cols-2;j++){
                int a=max({grid[i][j],grid[i][j+1],grid[i][j+2]});
                int b=max({grid[i+1][j],grid[i+1][j+1],grid[i+1][j+2]});
                int c=max({grid[i+2][j],grid[i+2][j+1],grid[i+2][j+2]});
                int d=max({a,b,c});
                temp.push_back(d);
            }
            res.push_back(temp);
        }
        return res;
    }
};

File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        rows,cols=len(grid),len(grid[0])
        res=[]
        for i in range(rows-2):
            temp=[]
            for j in range(cols-2):
                a=max([grid[i][j],grid[i][j+1],grid[i][j+2]])
                b=max([grid[i+1][j],grid[i+1][j+1],grid[i+1][j+2]])
                c=max([grid[i+2][j],grid[i+2][j+1],grid[i+2][j+2]])
                d=max([a,b,c])
                temp.append(d)
            res.append(temp)
        return res